Added build workflow and fixed build issues

patrickbussmann opened the pull request:
I had the same issue like #1.
So now I added GitHub build workflow and fixed the build.
It generates now a `VZD-Client-1.1.0.zip`.

But the script is skipping the tests at the moment because:
```
> Task :test

de.gematik.ti.epa.vzd.gem.testSuites.IntegrationTestsuite > de.gematik.ti.epa.vzd.gem.command.commandExecutions.AddDirEntryCertExecutionIntegrationTest.addCommandFailTest FAILED
    java.lang.ExceptionInInitializerError at AddDirEntryCertExecutionIntegrationTest.java:34
    java.lang.InstantiationError at AddDirEntryCertExecutionIntegrationTest.java:41

de.gematik.ti.epa.vzd.gem.testSuites.IntegrationTestsuite > de.gematik.ti.epa.vzd.gem.command.commandExecutions.AddDirEntryCertExecutionIntegrationTest.executeCommandExceptionTest FAILED
    java.lang.ExceptionInInitializerError at AddDirEntryCertExecutionIntegrationTest.java:34
    java.lang.InstantiationError at AddDirEntryCertExecutionIntegrationTest.java:41

de.gematik.ti.epa.vzd.gem.testSuites.IntegrationTestsuite > de.gematik.ti.epa.vzd.gem.command.commandExecutions.AddDirEntryCertExecutionIntegrationTest.addCommandSuccessTest FAILED
    java.lang.ExceptionInInitializerError at AddDirEntryCertExecutionIntegrationTest.java:34
    java.lang.InstantiationError at AddDirEntryCertExecutionIntegrationTest.java:41

de.gematik.ti.epa.vzd.gem.testSuites.IntegrationTestsuite > de.gematik.ti.epa.vzd.gem.command.commandExecutions.AddDirEntryExecutionIntegrationTest.addCommandDifferentTelematikIdTest FAILED
    java.lang.ExceptionInInitializerError at AddDirEntryExecutionIntegrationTest.java:34
    java.lang.InstantiationError at AddDirEntryExecutionIntegrationTest.java:41

de.gematik.ti.epa.vzd.gem.testSuites.IntegrationTestsuite > de.gematik.ti.epa.vzd.gem.command.commandExecutions.AddDirEntryExecutionIntegrationTest.addCommandFailTest FAILED
    java.lang.ExceptionInInitializerError at AddDirEntryExecutionIntegrationTest.java:34
    java.lang.InstantiationError at AddDirEntryExecutionIntegrationTest.java:41

de.gematik.ti.epa.vzd.gem.testSuites.IntegrationTestsuite > de.gematik.ti.epa.vzd.gem.command.commandExecutions.AddDirEntryExecutionIntegrationTest.executeCommandDifferentTelematikIdTest FAILED
    java.lang.ExceptionInInitializerError at AddDirEntryExecutionIntegrationTest.java:34
    java.lang.InstantiationError at AddDirEntryExecutionIntegrationTest.java:41

de.gematik.ti.epa.vzd.gem.testSuites.IntegrationTestsuite > de.gematik.ti.epa.vzd.gem.command.commandExecutions.AddDirEntryExecutionIntegrationTest.executeCommandExceptionTest FAILED
    java.lang.ExceptionInInitializerError at AddDirEntryExecutionIntegrationTest.java:34
    java.lang.InstantiationError at AddDirEntryExecutionIntegrationTest.java:41

de.gematik.ti.epa.vzd.gem.testSuites.IntegrationTestsuite > de.gematik.ti.epa.vzd.gem.command.commandExecutions.AddDirEntryExecutionIntegrationTest.addCommandSuccessTest FAILED
    java.lang.ExceptionInInitializerError at AddDirEntryExecutionIntegrationTest.java:34
    java.lang.InstantiationError at AddDirEntryExecutionIntegrationTest.java:41

de.gematik.ti.epa.vzd.gem.testSuites.IntegrationTestsuite > de.gematik.ti.epa.vzd.gem.command.commandExecutions.DeleteDirEntryCertExecutionIntegrationTest.delCertCommandFailTest FAILED
    java.lang.ExceptionInInitializerError at DeleteDirEntryCertExecutionIntegrationTest.java:34
    java.lang.InstantiationError at DeleteDirEntryCertExecutionIntegrationTest.java:41

de.gematik.ti.epa.vzd.gem.testSuites.IntegrationTestsuite > de.gematik.ti.epa.vzd.gem.command.commandExecutions.DeleteDirEntryCertExecutionIntegrationTest.delCertCommandSuccessTest FAILED
    java.lang.ExceptionInInitializerError at DeleteDirEntryCertExecutionIntegrationTest.java:34
    java.lang.InstantiationError at DeleteDirEntryCertExecutionIntegrationTest.java:41

de.gematik.ti.epa.vzd.gem.testSuites.IntegrationTestsuite > de.gematik.ti.epa.vzd.gem.command.commandExecutions.DeleteDirEntryCertExecutionIntegrationTest.executeCommandExceptionTest FAILED
    java.lang.ExceptionInInitializerError at DeleteDirEntryCertExecutionIntegrationTest.java:34
    java.lang.InstantiationError at DeleteDirEntryCertExecutionIntegrationTest.java:41

de.gematik.ti.epa.vzd.gem.testSuites.IntegrationTestsuite > de.gematik.ti.epa.vzd.gem.command.commandExecutions.DeleteDirEntryExecutionIntegrationTest.delCommandFailTest FAILED
    java.lang.ExceptionInInitializerError at DeleteDirEntryExecutionIntegrationTest.java:34
    java.lang.InstantiationError at DeleteDirEntryExecutionIntegrationTest.java:41

de.gematik.ti.epa.vzd.gem.testSuites.IntegrationTestsuite > de.gematik.ti.epa.vzd.gem.command.commandExecutions.DeleteDirEntryExecutionIntegrationTest.delCommandSuccessTest FAILED
    java.lang.ExceptionInInitializerError at DeleteDirEntryExecutionIntegrationTest.java:34
    java.lang.InstantiationError at DeleteDirEntryExecutionIntegrationTest.java:41

de.gematik.ti.epa.vzd.gem.testSuites.IntegrationTestsuite > de.gematik.ti.epa.vzd.gem.command.commandExecutions.DeleteDirEntryExecutionIntegrationTest.executeCommandEntryNotPresentTest FAILED
    java.lang.ExceptionInInitializerError at DeleteDirEntryExecutionIntegrationTest.java:34
    java.lang.InstantiationError at DeleteDirEntryExecutionIntegrationTest.java:41

de.gematik.ti.epa.vzd.gem.testSuites.IntegrationTestsuite > de.gematik.ti.epa.vzd.gem.command.commandExecutions.ModifyDirEntryExecutionIntegrationTest.modCommandDoAddTest FAILED
    java.lang.ExceptionInInitializerError at ModifyDirEntryExecutionIntegrationTest.java:33
    java.lang.InstantiationError at ModifyDirEntryExecutionIntegrationTest.java:40

de.gematik.ti.epa.vzd.gem.testSuites.IntegrationTestsuite > de.gematik.ti.epa.vzd.gem.command.commandExecutions.ModifyDirEntryExecutionIntegrationTest.modCommandSuccessTest FAILED
    java.lang.ExceptionInInitializerError at ModifyDirEntryExecutionIntegrationTest.java:33
    java.lang.InstantiationError at ModifyDirEntryExecutionIntegrationTest.java:40

de.gematik.ti.epa.vzd.gem.testSuites.IntegrationTestsuite > de.gematik.ti.epa.vzd.gem.command.commandExecutions.ModifyDirEntryExecutionIntegrationTest.executeCommandNotFoundTest FAILED
    java.lang.ExceptionInInitializerError at ModifyDirEntryExecutionIntegrationTest.java:33
    java.lang.InstantiationError at ModifyDirEntryExecutionIntegrationTest.java:40

de.gematik.ti.epa.vzd.gem.testSuites.IntegrationTestsuite > de.gematik.ti.epa.vzd.gem.command.commandExecutions.ReadDirEntryCertExecutionIntegrationTest.readCommandFailTest FAILED
    java.lang.ExceptionInInitializerError at ReadDirEntryCertExecutionIntegrationTest.java:34
    java.lang.InstantiationError at ReadDirEntryCertExecutionIntegrationTest.java:41

de.gematik.ti.epa.vzd.gem.testSuites.IntegrationTestsuite > de.gematik.ti.epa.vzd.gem.command.commandExecutions.ReadDirEntryCertExecutionIntegrationTest.executeCommandIdNotPresentTest FAILED
    java.lang.ExceptionInInitializerError at ReadDirEntryCertExecutionIntegrationTest.java:34
    java.lang.InstantiationError at ReadDirEntryCertExecutionIntegrationTest.java:41

de.gematik.ti.epa.vzd.gem.testSuites.IntegrationTestsuite > de.gematik.ti.epa.vzd.gem.command.commandExecutions.ReadDirEntryCertExecutionIntegrationTest.readCommandSuccessTest FAILED
    java.lang.ExceptionInInitializerError at ReadDirEntryCertExecutionIntegrationTest.java:34
    java.lang.InstantiationError at ReadDirEntryCertExecutionIntegrationTest.java:41

de.gematik.ti.epa.vzd.gem.testSuites.IntegrationTestsuite > de.gematik.ti.epa.vzd.gem.command.commandExecutions.ReadDirEntryExecutionIntegrationTest.readCommandFailTest FAILED
    java.lang.ExceptionInInitializerError at ReadDirEntryExecutionIntegrationTest.java:34
    java.lang.InstantiationError at ReadDirEntryExecutionIntegrationTest.java:41

de.gematik.ti.epa.vzd.gem.testSuites.IntegrationTestsuite > de.gematik.ti.epa.vzd.gem.command.commandExecutions.ReadDirEntryExecutionIntegrationTest.executeCommandIdNotPresentTest FAILED
    java.lang.ExceptionInInitializerError at ReadDirEntryExecutionIntegrationTest.java:34
    java.lang.InstantiationError at ReadDirEntryExecutionIntegrationTest.java:41

de.gematik.ti.epa.vzd.gem.testSuites.IntegrationTestsuite > de.gematik.ti.epa.vzd.gem.command.commandExecutions.ReadDirEntryExecutionIntegrationTest.readCommandSuccessTest FAILED
    java.lang.ExceptionInInitializerError at ReadDirEntryExecutionIntegrationTest.java:34
    java.lang.InstantiationError at ReadDirEntryExecutionIntegrationTest.java:41

de.gematik.ti.epa.vzd.gem.testSuites.IntegrationTestsuite > de.gematik.ti.epa.vzd.gem.command.commandExecutions.ExecutionBaseIntegrationTest.getUidTestMissMatchUid FAILED
    java.lang.ExceptionInInitializerError at ExecutionBaseIntegrationTest.java:32
    java.lang.InstantiationError at ExecutionBaseIntegrationTest.java:39

de.gematik.ti.epa.vzd.gem.testSuites.UnitTestsuite > de.gematik.ti.epa.vzd.gem.command.CommandsBuilderTest.checkParsingError FAILED
    org.junit.ComparisonFailure at CommandsBuilderTest.java:41

de.gematik.ti.epa.vzd.gem.testSuites.UnitTestsuite > de.gematik.ti.epa.vzd.gem.command.CommandsBuilderTest.doubledCommandIdDefinedTest FAILED
    java.lang.AssertionError at CommandsBuilderTest.java:59
        Caused by: de.gematik.ti.epa.vzd.gem.exceptions.ReadException at CommandsBuilderTest.java:59

92 tests completed, 26 failed

> Task :test FAILED
```

So `92 tests completed, 26 failed`.

Maybe someone can fix them and remove the ` -x test` from the `.github\workflows\build.yml` file.

Commit:
- Funktionality for add directory entries\n
- Funktionality for delete directory entries\n
- Funktionality for read directory entries\n

Commit:
- Funktionality for modify directory entries\n
- Funktionality for add certificates entries\n
- Funktionality for delete certificates entries\n
- Funktionality for read certificates entries\n

Commit:
Neue Funktionalitäten:
- Proxyfunktion
- Multithreading ready
- SaveModify (Modify ohne Überschreiben)

patrickbussmann committed:
Added build workflow and fixed build issues

Gematik-Entwicklung commented:
Hello @patrickbussmann ,

thanks for working on our product.

We tried unsuccessfully to recreate the behavior you described. It appears that the test server does not start successfully. The build environment may prevent a secondary process from starting. Could you check if the process starts and is up and running while executing the tests?

Please have a look at the [IntegrationTestsuite](https://github.com/gematik/app-VZD-Client/blob/master/src/test/java/de/gematik/ti/epa/vzd/gem/testSuites/IntegrationTestsuite.java) Class

Commit:
Features:
- Version des Clients wird beim Ausführen angegeben und ist über "-version" abzufragen

BugFixes:
- Suche nach nicht vorhandenem Eintrag führt nun zu Fehler
